test/nashorn/src/jdk/nashorn/api/scripting/test/ScriptObjectMirrorTest.java

```diff
@@ -282,7 +282,7 @@ public void conversionTest() throws ScriptException {
 
         ScriptObjectMirror obj = (ScriptObjectMirror)e.eval(
             "({ valueOf: function() { return 42 } })");
-        assertEquals(42.0, obj.to(Double.class));
+        assertEquals(new Double(42.0), obj.to(Double.class));
```

Comment:
```suggestion
        assertEquals(Double.valueOf(42.0), obj.to(Double.class));
```
`Double` constructor is deprecated since Java 9.